src/rust/engine/src/lib.rs

```diff
@@ -435,7 +435,7 @@ pub extern fn validator_run(
   scheduler_ptr: *mut RawScheduler,
   subject_types_ptr: *mut TypeId,
   subject_types_len: u64
-) -> *const Value {
+) -> Value {
```

Comment:
Tried it. I feel like we'd need to pull the thread further for it to be worth it. Not going to do it as part of this patch.